### PR TITLE
Common - Update `CBA_fnc_selectWeapon`

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Title.

### CODE REVIEW

**Review by PabstMirror — APPROVED** [review]

**Comment by johnb432** [comment]
Hold on merging for a second.

**Comment by johnb432** [comment]
When passing `""` as `_weapon`, it holsters the weapon. This updated version doesn't, I'll need to take care of that.

**Comment by johnb432** [comment]
@PabstMirror I believe that should have done the trick.

File: addons/common/fnc_selectWeapon.sqf [hunk 1]
```diff
@@ -23,30 +23,45 @@ Examples:
...
+        _weaponState set [2, _mode];
     };
+
+    _unit selectWeapon _weaponState // return
 } else {
-    while {_index < MAXIMUM_TRIES && {currentMuzzle _unit != _weapon || {currentWeaponMode _unit != _mode}}} do {
-        _unit action ["SwitchWeapon", _vehicle, _unit, _index];
-        _index = _index + 1;
+    private _turretPath = _vehicle unitTurret _unit;
+    (weaponState [_vehicle, _turretPath, _weapon]) params ["", "_muzzle", "_currentMode"];
```
Comment (PabstMirror) [inline addons/common/fnc_selectWeapon.sqf:55]:
```suggestion
    (weaponState [_vehicle, _turretPath, _weapon]) params ["_weapon", "_muzzle", "_currentMode"];
```
need this for selecting muzzles
e.g.
`[player, "AP"] call cba_fnc_selectWeapon` on badger IFV
want `_weapon` updated so it does
`selectWeaponTurret ["autocannon_40mm_CTWS",[0],"AP","player"]`

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**